htsohm/htsohm.py

```diff
@@ -43,61 +37,64 @@ def queue_all_materials(run_id, generation, queue):
+
+def seed_generation(run_id, children_per_generation, number_of_atomtypes):
     generation = 0
-    init_materials_in_database(run_id, children_per_generation, generation)
     write_seed_definition_files(run_id, children_per_generation, number_of_atomtypes)
-    if queue is not None:
-        queue_all_materials(run_id, generation, queue)
-    else:
-        simulate_all_materials(run_id, generation)
+    session.commit()
 
-def next_generation(run_id, children_per_generation, generation, queue=None):
-    init_materials_in_database(run_id, children_per_generation, generation)
-    screen_parents(run_id, children_per_generation, generation)
-    create_next_generation(run_id, generation)
-    if queue is not None:
-        queue_all_materials(run_id, generation, queue)
-    else:
-        simulate_all_materials(run_id, generation)
+def update_strength_array(run_id, generation):
+    if generation == 1:
+        create_strength_array(run_id)
+    elif generation >= 2:
+        recalculate_strength_array(run_id, generation)
 
 def htsohm(children_per_generation,    # number of materials per generation
            number_of_atomtypes,        # number of atom-types per material
            strength_0,                 # intial strength parameter
            number_of_bins,             # number of bins for analysis
            max_generations=20,         # maximum number of generations
-           acceptance_value=-0.5):      # desired degree of `convergence`
+           dummy_test_trials=1,        # number of re-simulations for dummy-test
+           acceptance_value=-0.005):      # desired degree of `convergence`
     ############################################################################
     # write run-configuration file
     run_id = write_config_file(children_per_generation, number_of_atomtypes, strength_0,
-        number_of_bins, max_generations)["run-id"]
+        number_of_bins, max_generations, dummy_test_trials, acceptance_value)["run-id"]
 
     convergence = acceptance_value + 1          # initialize convergence with arbitrary value
     for generation in range(max_generations):
         while convergence >= acceptance_value:
             if generation == 0:                     # SEED GENERATION
                 seed_generation(run_id, children_per_generation, number_of_atomtypes)
+                simulate_all_materials(run_id, generation)
                 convergence = evaluate_convergence(run_id)
                 save_convergence(run_id, generation, convergence)
             elif generation >= 1:                   # FIRST GENERATION, AND ON...
-                next_generation(run_id, children_per_generation, generation)
+                update_strength_array(run_id, generation)
+                create_next_generation(run_id, generation)
+                simulate_all_materials(run_id, generation)
                 convergence = evaluate_convergence(run_id)
                 save_convergence(run_id, generation, convergence)
```

Comment:
This isn't part of these commits, but I see there is convergence here, but not in manage_run, so nothing on the cluster will run the convergence code. 

We should think about this a little bit more. Maybe the code that manages the hpc and one cycle of htsohm() can be combined so that there isn't this code duplication (and the likelihood of missing something).